integrations/apify/src/client.ts

```diff
@@ -0,0 +1,435 @@
+          this.logger.forBot().info(`[GET AND SYNC RUN RESULTS] KB indexing completed. Documents indexed: ${filesCreated}`);
+        } else {
+          this.logger.forBot().info(`[GET AND SYNC RUN RESULTS] Starting file sync to default path: ${targetPath}`);
+          filesCreated = await this.syncContentToBotpress(allItems, targetPath, kbId || "kb-default");
+          this.logger.forBot().info(`[GET AND SYNC RUN RESULTS] File sync completed. Files created: ${filesCreated}`);
+        }
+      } catch (error) {
+        this.logger.forBot().error('Error syncing content to Botpress:', error);
+        return {
+          success: false,
+          message: 'Error syncing content to Botpress',
+          data: {
+            error: error instanceof Error ? error.message : 'Error syncing content to Botpress',
+          },
+        };
+      }
+
+      return {
+        success: true,
+        message: 'Run results retrieved successfully',
+        data: {
+          runId: run.id,
+          datasetId: run.defaultDatasetId,
+          itemsCount: allItems.length,
+          filesCreated,
+          syncTargetPath: targetPath,
+        },
+      };
+    } catch (error) {
+      this.logger.forBot().error('Error getting run information:', error);
+      
+      if (error instanceof ApifyApiError) {
+        const { message, type, statusCode, clientMethod } = error;
+        this.logger.forBot().error('Apify API Error:', { message, type, statusCode, clientMethod });
+        
+        return {
+          success: false,
+          message: `Apify API Error: ${message}`,
+          data: {
+            error: message,
+            type,
+            statusCode,
+            clientMethod,
+          },
+        };
+      }
+
+      return {
+        success: false,
+        message: error instanceof Error ? error.message : 'Error in getAndSyncRunResults',
+        data: {
+          error: error instanceof Error ? error.message : 'Error in getAndSyncRunResults',
+        },
+      };
+    }
+  }
+
+  private isLastPage(offset: number, limit: number, total: number): boolean {
+    return offset + limit >= total;
+  }
+
+  private getNextOffset(offset: number, limit: number): number {
+    return offset + limit;
+  }
+
+  private async fetchAllDatasetItems(dataset: any) {
+    let allItems: {
+      markdown?: string;
+      html?: string;
+      text?: string;
+      url?: string;
+      metadata?: { url?: string };
+    }[] = [];
+    let offset = 0;
+    const limit = 1000; 
+
+    while (true) {
+      const { items, total } = await dataset.listItems({ limit, offset });
+      
+      this.logger.forBot().info(`Fetched ${items.length} items (offset: ${offset}, total: ${total})`);
+      
+      allItems.push(...items);
+      
+      // If there are no more items to fetch, exit the loading
+      if (this.isLastPage(offset, limit, total)) {
+        break;
+      }
+      
+      offset = this.getNextOffset(offset, limit);
```

Comment:
question: have you tried this on a large source? i'm concerned about timeouts. This is running on max 60s lambdas.